feat: add `--send` to `contract invoke` and default to only sending transaction if there are writes in the ledger footprint (#1518)
stellar/stellar-cli: FULL_HELP_DOCS.md

###### **Options:**

* `--id <CONTRACT_ID>` — Contract ID to invoke

> * `--is-view` — View the result simulating and do not sign and submit transaction. Ieprecated use `--send=no`

Reviewer: ```suggestion
* `--is-view` — View the result simulating and do not sign and submit transaction. Deprecated use `--send=no`
```

> * `--is-view` — View the result simulating and do not sign and submit transaction. Ieprecated use `--send=no`

Reviewer 2: sp: `Deprecated`

Author: Yeah this is why we shouldn't check in this file. I updated the source code where these comments come from by accepting a suggestion from Leigh, but then I need to re generate this file locally...

Reviewer: ```suggestion
* `--is-view` — View the result simulating and do not sign and submit transaction. Deprecated use `--send=no`
```

Reviewer: Git history shows it being fixed in the code, then being changed back to `Ieprecated` in the code. Automation won't prevent us from undoing typo fixes in the code. I don't think we need more automation for this.

I'm going to push back on introducing more complexity for the sake of having a simply accessible help file that we can review in PRs as a way to understanding the UX/UI clearly.

Author: Fair enough.

* `--instructions <INSTRUCTIONS>` — Number of instructions to simulate
* `--build-only` — Build the transaction and only write the base64 xdr to stdout
* `--sim-only` — Simulate the transaction and only write the base64 xdr to stdout
* `--send <SEND>` — Whether or not to send a transaction

  Default value: `if-write`

  Possible values:

>   - `if-write`:
>     Only send transaction if there are ledger writes, otherwise return simulation result
>   - `no`:
>     Do not send transaction, return simulation result
>   - `yes`:
>     Always send transaction

Reviewer: ```suggestion
  - `if-write`:
    Send the transaction to the network if there are ledger writes or events published, otherwise output the simulation result and do not send to the network
  - `no`:
    Do not send the transaction to the network, and output the simulation result
  - `yes`:
    Send the transaction to the network, and output the transaction result
```
